UsbIpServer/ConnectedClient.cs

```diff
@@ -130,16 +130,21 @@ async Task HandleRequestImportAsync(CancellationToken cancellationToken)
                     using var attachedClientTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
                     Watcher.WatchDevice(busid, () => attachedClientTokenSource.Cancel());
                     var attachedClientToken = attachedClientTokenSource.Token;
+                    RegistryUtils.SetDeviceAsAttached(exportedDevice);
+                    var iPEndPoint = ClientContext.TcpClient.Client.RemoteEndPoint as IPEndPoint;
+                    RegistryUtils.SetDeviceAddress(exportedDevice, iPEndPoint!.Address.MapToIPv4().ToString());
                     await ServiceProvider.GetRequiredService<AttachedClient>().RunAsync(attachedClientToken);
                 }
                 finally
                 {
                     Watcher.StopWatchingDevice(busid);
+                    RegistryUtils.SetDeviceAsDetached(exportedDevice);
                     Logger.LogInformation(LogEvents.ClientDetach, $"Client {ClientContext.TcpClient.Client.RemoteEndPoint} released device at {exportedDevice.BusId} ({exportedDevice.Path}).");
                 }
             }
-            catch
+            catch (Exception ex)
             {
+                Logger.LogError(LogEvents.ClientError, $"An exception occurred while communicating with the client: {ex}");
```

Comment:
This path is also followed for EndOfStreamException, which is *normal* client hangup. This will log an error for every normal detach.

Reply:
fixed